### ISSUE TITLE
Add support to auto populate @throws statements

### ISSUE THREAD

**Issue by narendravaghela** [nikhilkalige/docblockr#227]

**Comment by narendravaghela** [comment]
@MoritzKn I assume this feature will be available for PHP language as well, right?

**Comment by MoritzKn** [comment]
Are throw statements documented in PHP like in Java? if so, yes I would make sense to add this feature for PHP too.

**Comment by narendravaghela** [comment]
Yes, PHP does it. For example, see below class and function documentation.

```php
<?php

namespace App\Controller;

use Cake\Core\Configure;
use Cake\Network\Exception\NotFoundException;
use Cake\View\Exception\MissingTemplateException;

/**
 * Static content controller
 *
 * This controller will render views from Template/Pages/
 *
 * @link http://book.cakephp.org/3.0/en/controllers/pages-controller.html
 */
class PagesController extends AppController
{

    /**
     * Displays a view
     *
     * @return void|\Cake\Network\Response
     * @throws \Cake\Network\Exception\NotFoundException When the view file could not
     *   be found or \Cake\View\Exception\MissingTemplateException in debug mode.
     */
    public function display()
    {
        $path = func_get_args();

        try {
            $this->render(implode('/', $path));
        } catch (MissingTemplateException $e) {
            if (Configure::read('debug')) {
                throw $e;
            }
            throw new NotFoundException();
        }
    }
}
```

**Comment by MoritzKn** [comment]
@narendravaghela Ok, this could get complicated, because docblockr only parses the function signature. In Java checked exceptions are declared in the method signature like so:
```java
void foo() throws IOException {}
```
But in your PHP code the exceptions is thrown without being declared before. To find all executions docblockr would have to parse the method body.
In any way I don't think I will find time for this. I you are willing to implement this yourself I'm happy to help; review and merge your PR.

**Comment by narendravaghela** [comment]
Thanks @MoritzKn 
Actually I do not have much knowledge of programming in NodeJs. I will try if I can do something in this for PHP, otherwise you can consider Java platform as of now.

**Comment by MoritzKn** [comment]
In #156 it turned out that automatically adding `@throws` attributes was already implemented for Java, but wasn't working correctly. I fixed this in order to fix #156. So for Java this works now.
I will close this issue, because I think implementing this for PHP is out of scope (as explained above).
